src/main/java/org/fcrepo/importexport/importer/Importer.java

```diff
@@ -89,49 +99,84 @@ private FcrepoClient client() {
-            final Model model = parseFile(f);
-            final ResIterator binaryResources = model.listResourcesWithProperty(RDF_TYPE, NON_RDF_SOURCE);
-            if (binaryResources.hasNext()) {
-                uri = new URI(binaryResources.nextResource().getURI());
-                logger.info("Importing binary {}", f.getAbsolutePath());
-                response = importBinary(uri, sanitize(model));
-            } else {
-                uri = uriForFile(f, config.getDescriptionDirectory());
-                logger.info("Importing container {}", f.getAbsolutePath());
-                response = importContainer(uri, sanitize(model));
-            }
-
-            if (response.getStatusCode() == 401) {
-                throw new AuthenticationRequiredRuntimeException();
-            } else if (response.getStatusCode() > 204 || response.getStatusCode() < 200) {
-                logger.warn("Error while importing {} ({}): {}",
-                   f.getAbsolutePath(), response.getStatusCode(), IOUtils.toString(response.getBody()));
-            } else {
-                logger.info("Imported {}: {}", f.getAbsolutePath(), uri);
+        // The path, relative to the metadata root in the export directory.
+        // This used in place of the full path to make the output more readable.
+        final String sourceRelativePath = config.getDescriptionDirectory().toPath().relativize(f.toPath()).toString();
+
+        if (f.getPath().endsWith(BINARY_EXTENSION)) {
+            // ... this is only expected to happens when binaries and metadata are written to the same directory...
+            logger.debug("Skipping binary {}: it will be imported when its metadata is imported.", sourceRelativePath);
+            return;
+        } else if (!f.getPath().endsWith(config.getRdfExtension())) {
+            // this could be hidden files created by the OS
+            logger.info("Skipping file with unexpected extension ({}).", sourceRelativePath);
+            return;
+        } else {
+            FcrepoResponse response = null;
+            URI destinationUri = null;
+            try {
+                final Model model = parseFile(f);
+                if (model.contains(null, RDF_TYPE, createResource(REPOSITORY_NAMESPACE + "RepositoryRoot"))) {
+                    logger.debug("Skipping import of repository root.");
+                    return;
+                }
+                final ResIterator binaryResources = model.listResourcesWithProperty(RDF_TYPE, NON_RDF_SOURCE);
+                if (binaryResources.hasNext()) {
+                    destinationUri = new URI(binaryResources.nextResource().getURI());
+                    logger.info("Importing binary {}", sourceRelativePath);
+                    response = importBinary(destinationUri, sanitize(model));
+                } else {
+                    destinationUri = new URI(config.getResource().toString()
+                            + (config.getResource().toString().endsWith("/") ? "" : "/")
```

Comment:
Oops... that said, it looks like there's explicit logic in the setResource to ensure there's NOT a trailing slash.  I'll just add one without the inline logic.